cub/test/catch2_test_memcpy_batched.cu

```diff
@@ -0,0 +1,201 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cub/device/device_memcpy.cuh>
+#include <cub/util_macro.cuh>
+
+#include <thrust/copy.h>
+#include <thrust/detail/raw_pointer_cast.h>
+#include <thrust/fill.h>
+#include <thrust/gather.h>
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/random.h>
+#include <thrust/reduce.h>
+#include <thrust/scatter.h>
+#include <thrust/sequence.h>
+#include <thrust/shuffle.h>
+#include <thrust/transform.h>
+
+#include "catch2_test_launch_helper.h"
+#include <c2h/catch2_test_helper.h>
+
+// %PARAM% TEST_LAUNCH lid 0:1:2
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceMemcpy::Batched, memcpy_batched);
+
+/**
+ * @brief Function object that takes an offset and returns an iterator at the given
+ * offset relative to a fixed base iterator.
+ */
+template <typename IteratorT>
+struct offset_to_ptr_op
+{
+  IteratorT base_it;
+
+  template <typename T>
+  __host__ __device__ __forceinline__ IteratorT operator()(T offset) const
+  {
+    return base_it + offset;
+  }
+};
+
+/**
+ * @brief Used for generating a shuffled but cohesive sequence of output-buffer offsets for the
+ * sequence of input-buffers.
+ */
+template <typename BufferOffsetT, typename BufferSizeT, typename ByteOffsetT>
+void get_shuffled_buffer_offsets(const c2h::device_vector<BufferSizeT>& buffer_sizes,
+                                 c2h::device_vector<ByteOffsetT>& buffer_offsets,
+                                 c2h::seed_t seed)
```

Comment:
Suggestion: Avoid out parameters:
```suggestion
template <typename BufferOffsetT, typename BufferSizeT, typename ByteOffsetT>
auto get_shuffled_buffer_offsets(const c2h::device_vector<BufferSizeT>& buffer_sizes,
                                 c2h::seed_t seed) -> c2h::device_vector<ByteOffsetT>
```

Reply:
Done 👍  _...at the cost of one extra explicit template argument_